src/client/csfle/client_encryption.rs

```diff
@@ -182,7 +207,29 @@ impl ClientEncryption {
 
     /// Encrypts a BsonValue with a given key and algorithm.
     /// Returns an encrypted value (BSON binary of subtype 6).
-    pub async fn encrypt(&self, value: bson::RawBson, opts: EncryptOptions) -> Result<Binary> {
```

Comment:
One slight drawback of this approach is that we no longer see the return value in the method declaration. This isn't totally unprecedented though, since many of tokio's methods return a named future type. Would there be any value in introducing a sealed trait for our various Action types that indicates the output type?

Reply:
The drawback of introducing a trait is that clients would have to import that trait; for now I've gone with adding specific documentation on return types.